src/data/automation_i18n.ts

```diff
@@ -488,24 +494,68 @@ const tryDescribeTrigger = (
         hours < 0 ||
         (hours_interval && hours === 0)
       ) {
-        return "Invalid Time Pattern Hours";
+        invalidParts.push("hours");
       }
 
       if (hours_all || (hours_interval && hours === 1)) {
-        result += "every hour";
+        hoursChoice = "every";
       } else if (hours_interval) {
-        result += `every ${hours} hours`;
+        hoursChoice = "every_interval";
       } else {
-        result += `${
+        hoursChoice =
           trigger.seconds !== undefined || trigger.minutes !== undefined
-            ? ""
-            : "on"
-        } the ${hours}${ordinalSuffix(hours)} hour`;
+            ? "has_seconds_or_minutes"
+            : "on_the_xth";
       }
     } else {
-      result += "every hour";
+      hoursChoice = "every";
     }
-    return result;
+
+    if (invalidParts.length !== 0) {
+      return hass.localize(
+        `${triggerTranslationBaseKey}.time_pattern.description.invalid`,
+        {
+          parts: formatListWithAnds(
+            hass.locale,
+            invalidParts.map((invalidPart) =>
+              hass.localize(
+                `${triggerTranslationBaseKey}.time_pattern.${invalidPart}`
+              )
+            )
+          ),
+        }
+      );
+    }
+
+    return hass.localize(
+      `${triggerTranslationBaseKey}.time_pattern.description.full`,
+      {
+        secondsChoice: secondsChoice,
+        minutesChoice: minutesChoice,
+        hoursChoice: hoursChoice,
+        seconds: seconds,
+        minutes: minutes,
+        hours: hours,
+        secondsWithOrdinal: hass.localize(
+          `${triggerTranslationBaseKey}.time_pattern.description.ordinal`,
+          {
+            part: seconds,
```

Comment:
This should be using `new Intl.PluralRules("en", {type: "ordinal"}).select(seconds)`